### PR TITLE
Add SPI for native for Error Code and Exception Name providers

### LINKED ISSUE #27414: Smallrye GraphQL native: SPI ErrorExtensionProvider implementations not included
### Describe the bug

I use quarkus-smallrye-graphql on top op quarkus 2.11.2.Final. Since I upgraded from 2.9.2.Final my integration tests started to fail: I do have ITs that make the application throw an exception, for example, when validation fails, or writing to the database fails. Those tests now fail due to a timeout on the GQL client end. On the server side I see the following exception:

`ServiceConfigurationError: failed to load io.smallrye.graphql.api.ErrorExtensionProvider, class io.smallrye.graphql.execution.error.ErrorCodeExtensionProvider does not exist`

### Expected behavior

Application should fail with expected (gracefully handled) exception.

### Actual behavior

Application does not close response which leads to timeout on client side.

### How to Reproduce?

1. Create Quarkus Application with SmallRye GraphQL extension
2. Annotate at least one method with `@Mutation` that throws an exception
3. Execute that mutation (via HTTP client or generated GQL client)

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.11.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

### Additional information

`smallrye-graphql-1.6.0.jar` comes with `META-INF/services/io.smallrye.graphql.api.ErrorExtensionProvider`, the classes referenced (`io.smallrye.graphql.execution.error.ErrorCodeExtensionProvider` and `io.smallrye.graphql.execution.error.ExceptionNameErrorExtensionProvider`) do not end up in the native binary, however.

If I manually add the following entries to my `reflect-config.json`, everything works as expected again:
```json
{
  "name":"io.smallrye.graphql.execution.error.ErrorCodeExtensionProvider",
  "methods":[{"name":"<init>","parameterTypes":[] }]
},
{
  "name":"io.smallrye.graphql.execution.error.ExceptionNameErrorExtensionProvider",
  "methods":[{"name":"<init>","parameterTypes":[] }]
}
```

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix #27414

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Review by jmartisk — COMMENTED** [review]
The fix LGTM, but to point out that if the goal was to fix a native-mode-specific problem, then the provided test doesn't help verify that, as unit tests don't run in native mode, you'd need to do this in `integration-tests`. I'll leave it up to you if you want to change it or not

**Comment by phillip-kruger** [comment]
Yea hold on let me change that. I assume all tests run native too.

**Review by jmartisk — APPROVED** [review]

**Pull request merged**
